tasks/post_copy_request_to_queue/post_copy_request_to_queue.py

```diff
@@ -0,0 +1,228 @@
+        bucket_name = record["s3"]["bucket"]["name"]
+
+    sql = """
+        SELECT
+            job_id, granule_id, filename, restore_destination
+        FROM
+            orca_recoverfile
+        WHERE
+            key_path = %s
+        AND
+            status_id = %d
+    """
+    # Gets the dbconnection info.
+    # query the db table
+    try:
+        db_connect_info = requests_db.get_dbconnect_info()
+        rows = database.single_query(sql, db_connect_info, (key_path, OrcaStatus.PENDING.value))
+        if len(rows) == 0:
+            message = f"No metadata found for {key_path}"
+            LOGGER.fatal(message)
+            raise Exception(message)
+        # # convert db result to json
+        db_result_json = database.result_to_json(rows)
+
+    except Exception as ex:
+        message = f"Unable to retrieve {key_path} metadata"
+        LOGGER.error(message)
+        raise ex(message)
+
+    # grab the parameters from the db in json format.
+    job_id = db_result_json["job_id"]
+    granule_id = db_result_json["granule_id"]
+    filename = db_result_json["filename"]
+    restore_destination = db_result_json["restore_destination"]
+
+    my_base_delay = retry_sleep_secs
+    # define the input body for copy_files_to_archive lambda
+    new_data = {
+        "job_id": job_id,
+        "granule_id": granule_id,
+        "filename": filename,
+        "source_key": key_path,
+        "target_key": key_path,  # todo add a card to configure target_key in the future
+        "restore_destination": restore_destination,
+        "source_bucket": bucket_name,
+    }
+    # post to recovery queue. Retry using exponential delay if it fails
+    for retry in range(max_retries):
+        try:
+            post_entry_to_queue(
+                "orca_recoveryfile", new_data, RequestMethod.NEW, recovery_queue_url
+            )
+        except Exception:
+            LOGGER.error(
+                f"Ran into error posting to SQS {recovery_queue_url} {retry+1} time(s)"
+            )
+            my_base_delay = exponential_delay(my_base_delay, retry_backoff)
+            continue
+        break
+    else:
+        message = f"Error sending message to {recovery_queue_url} for {new_data}"
+        logging.critical(message)
+        raise Exception(message)
+
+    # resetting my_base_delay
+    my_base_delay = retry_sleep_secs
+    # post to DB-queue. Retry using exponential delay if it fails
+    for retry in range(max_retries):
+        try:
+            post_status_for_file_to_queue(
+                job_id,
+                granule_id,
+                filename,
+                key_path,
+                restore_destination,
+                OrcaStatus.STAGED,
+                None,
+                RequestMethod.UPDATE,
+                db_queue_url,
+            )
+        except Exception:
+            LOGGER.error(
+                f"Ran into error posting to SQS {db_queue_url} {retry+1} time(s)"
```

Comment:
Recommend including the exception in the message.

Reply:
will change to:

except Exception as ex:
LOGGER.error(
f"Ran into error posting to SQS {db_queue_url} {retry+1} time(s) with exception {ex}"
)